apps/04TowersOfHanoi.js

```diff
@@ -19,24 +19,45 @@ function printStacks() {
   console.log("c: " + stacks.c);
 }
 
-function movePiece() {
-  // Your code here
 
-}
 
-function isLegal() {
-  // Your code here
+// pop and push to move pieces from one stack to another.
+function movePiece(startStack, endStack) {
+  var piece = stacks[startStack].pop();
+  stacks[endStack].push(piece);
+}
 
+// Check if endStack is empty.
+function isLegal(startStack, endStack) {
+  if (stacks[endStack].length < 1) {
+    return true;
+  }
+  // Check if the number in endStack is smaller than the piece being moved. If so, it is a legal move.
+  else if ((stacks[startStack][(stacks[startStack].length - 1)]) < (stacks[endStack][(stacks[endStack].length - 1)])) {
```

Comment:
It might be worth some cleanup by adding intermediate variables here like you did in `movePiece`